### PR TITLE
Add test cases

### PR DESCRIPTION
**Pull request by Tomoko-hjf**
### PR Docs
https://github.com/PaddlePaddle/docs/pull/5924
### PR APIs
Others

### CODE REVIEW

**Comment by zhwesky2010** [comment]
处理下冲突

File: paconvert/api_matcher.py [hunk 1]
```diff
@@ -3734,3 +3734,64 @@ def get_paddle_class_nodes(self, func, args, kwargs):
...
+                return "{}({})".format(
+                    self.get_paddle_api(), self.kwargs_to_str(kwargs)
+                )
+            else:
+                return None
+        return "{}({})".format(self.get_paddle_api(), self.kwargs_to_str(kwargs))
+
+
+class UnpoolMatcher(BaseMatcher):
+    def generate_code(self, kwargs):
```
Comment (zhwesky2010) [inline paconvert/api_matcher.py:3766]: 同文档中问题，如果不astype能运行吗

File: paconvert/api_matcher.py [hunk 2]
```diff
@@ -3734,3 +3734,64 @@ def get_paddle_class_nodes(self, func, args, kwargs):
...
+        kwargs["indices"] = (
+            "(" + kwargs.pop("indices").strip("\n") + ").astype('int32')"
+        )
+
+        return GenericMatcher.generate_code(self, kwargs)
+
+
+class LstsqMatcher(BaseMatcher):
```
Comment (zhwesky2010) [inline paconvert/api_matcher.py:3774]: 感觉这些地方的差异较小，只有一个rank，一般不会有什么影响，就还是不用astype吧。单测里可以跳过这个rank.dtype的检查

File: paconvert/api_matcher.py [hunk 3]
```diff
@@ -3734,3 +3734,64 @@ def get_paddle_class_nodes(self, func, args, kwargs):
 
         self.write_aux_code()
         return "unchange"
+
+
+class FunctionalAdaptiveMaxPoolMatcher(BaseMatcher):
```
Comment (zhwesky2010) [inline paconvert/api_matcher.py:3739]: 这个差异可能也较小，因为网络中的还是out1，看看out2如果int32会不会有什么大影响吧。如果对深度学习训练影响不大，一般也可忽略。

File: paconvert/api_matcher.py (hunk 1 above)
Comment (Tomoko-hjf) [reply paconvert/api_matcher.py:3766]: 不可以运行，目前只支持`int32`.
image

File: paconvert/api_matcher.py (hunk 2 above)
Comment (Tomoko-hjf) [reply paconvert/api_matcher.py:3774]: done

File: paconvert/api_matcher.py (hunk 3 above)
Comment (zhwesky2010) [reply paconvert/api_matcher.py:3739]: 就还是放到API功能问题里去修吧

File: paconvert/api_matcher.py (hunk 2 above)
Comment (Tomoko-hjf) [reply paconvert/api_matcher.py]: 这个已跳过检查 `dtype`

File: paconvert/api_matcher.py (hunk 3 above)
Comment (Tomoko-hjf) [reply paconvert/api_matcher.py]: 这个已跳过检查 `dtype`，并记录。

File: paconvert/api_mapping.json [hunk 4]
```diff
@@ -1,6 +1,5 @@
 {
   "torch.nn.functional.log_softmax": {
-    "Matcher": "GenericMatcher",
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:3]: 这个是什么原因删掉

File: paconvert/api_mapping.json [hunk 5]
```diff
@@ -3542,7 +3541,10 @@
     ],
     "kwargs_change": {
       "return_indices": "return_mask"
-    }
+    },
+    "unsupport_args": [
+      "return_indices"
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:3546]: 这个是什么功能不支持

File: paconvert/api_mapping.json [hunk 6]
```diff
@@ -3873,7 +3881,7 @@
     }
   },
   "torch.nn.LogSoftmax": {
-    "Matcher": "GenericMatcher",
+    "Matcher": "SoftmaxMatcher",
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:3884]: 这个是如果有dim就不支持吗

File: tests/test_min.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    obj.run(
+        pytorch_code,
+        ["out"],
+        unsupport=True,
+        reason="paddle does not return index when dim is specified",
+    )
+
+
+# paddle does not return index when dim is specified
+def _test_case_7():
```
Comment (zhwesky2010) [inline tests/test_min.py:109]: 这个不用unsupport=True吗

File: tests/test_nn_AdaptiveMaxPool1d.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                            [-0.8833, -0.1369, -0.0168, -0.5409, -0.1511, -0.1240, -1.1870, -1.8816]]])
+        model = nn.AdaptiveMaxPool1d(output_size=5, return_indices=True)
+        result = model(x)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
+        reason="When index is returned, the type of paddle is int32 and torch is int64",
```
Comment (zhwesky2010) [inline tests/test_nn_AdaptiveMaxPool1d.py:65]: return_indices不支持的原因是什么

File: tests/test_nn_LogSoftmax.py [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+                        [6.0, 7.0, 8.0, 9.0]]])
+        model = nn.LogSoftmax()
+        result = model(x)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
+        reason="When dim is None, paddle and pytorch generate different results",
```
Comment (zhwesky2010) [inline tests/test_nn_LogSoftmax.py:77]: 这个是什么原因，api有问题吗

File: tests/test_nn_Softmax.py [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+                        [6.0, 7.0, 8.0, 9.0]]])
+        model = nn.Softmax()
+        result = model(x)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
+        reason="When dim is None, paddle and pytorch generate different results",
```
Comment (zhwesky2010) [inline tests/test_nn_Softmax.py:77]: 这个是什么原因，api有问题吗

File: tests/test_nn_functional_log_softmax.py [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+                            [[1.0, -2.0, -3.0, 4.0],
+                            [-5.0, 6.0, 7.0, -8.0],
+                            [6.0, 7.0, 8.0, 9.0]]])
+        result = F.log_softmax(x)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
```
Comment (zhwesky2010) [inline tests/test_nn_functional_log_softmax.py:39]: 这个是什么原因

File: tests/test_nn_functional_log_softmax.py [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+                            [[1.0, -2.0, -3.0, 4.0],
+                            [-5.0, 6.0, 7.0, -8.0],
+                            [6.0, 7.0, 8.0, 9.0]]])
+        result = F.log_softmax(x, dim=1)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
```
Comment (zhwesky2010) [inline tests/test_nn_functional_log_softmax.py:61]: 这个是什么原因

File: tests/test_nn_functional_log_softmax.py [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+                            [[1.0, -2.0, -3.0, 4.0],
+                            [-5.0, 6.0, 7.0, -8.0],
+                            [6.0, 7.0, 8.0, 9.0]]])
+        result = F.log_softmax(x, dtype=torch.float64)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
```
Comment (zhwesky2010) [inline tests/test_nn_functional_log_softmax.py:83]: 这个是什么原因

File: tests/test_nn_functional_log_softmax.py [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+                            [[1.0, -2.0, -3.0, 4.0],
+                            [-5.0, 6.0, 7.0, -8.0],
+                            [6.0, 7.0, 8.0, 9.0]]])
+        result = F.log_softmax(x, _stacklevel=2)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
```
Comment (zhwesky2010) [inline tests/test_nn_functional_log_softmax.py:105]: 这个api是什么问题

File: tests/test_nn_Softmax.py (hunk 10 above)
Comment (zhwesky2010) [reply tests/test_nn_Softmax.py:77]: 这个dim的设置，是因为paddle为-1和竞品不同吗？没办法通过Matcher来写吗

File: tests/test_min.py (hunk 7 above)
Comment (zhwesky2010) [reply tests/test_min.py:109]: 这个看起来是Matcher的问题啊

File: tests/test_nn_Softmax.py (hunk 10 above)
Comment (Tomoko-hjf) [reply tests/test_nn_Softmax.py:77]: 是的，竞品默认值为None，paddle默认值是-1 。paddle如果设置为None，会报错不支持None，这样的话就没办法通过Matcher写吧

File: tests/test_min.py (hunk 7 above)
Comment (Tomoko-hjf) [reply tests/test_min.py:109]: 是Matcher的问题，但是在Matcher中无法准确区分第二个参数代表的是要比较的`Tensor`还是指定`dim`吧，他们都可以是一个变量名

File: tests/test_min.py [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+        import torch
+        x = torch.tensor([[1, 2, 3], [3, 4, 6]])
+        result = torch.min(x, dim=1)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
+        reason="paddle does not return index when dim is specified",
```
Comment (zhwesky2010) [inline tests/test_min.py:45]: 这些地方也要优化一下Matcher，指定了dim时，用min/argmin组合就可以

test_min/test_max的文档、Matcher、单测可以同步一起修复

File: tests/test_min.py [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+    )
+
+
+# paddle does not return index when dim is specified
+def _test_case_7():
+    pytorch_code = textwrap.dedent(
+        """
+        import torch
+        dim, keepdim = 1, False
+        result = torch.min(torch.tensor([[1, 2, 3], [3, 4, 6]]), dim, keepdim)
```
Comment (zhwesky2010) [inline tests/test_min.py:114]: 这个可以优化下Matcher，torch.min(input, other, *, out=None) 这种用法时，只能是两个参数，如果是3个参数，那么第3个必然是out

File: tests/test_min.py (hunk 15 above)
Comment (Tomoko-hjf) [reply tests/test_min.py]: done

File: tests/test_min.py (hunk 16 above)
Comment (Tomoko-hjf) [reply tests/test_min.py]: done

**Comment by Tomoko-hjf** [comment]
构建镜像时出现了问题，导致CI没通过
image

File: paconvert/api_matcher.py [hunk 17]
```diff
@@ -392,37 +392,72 @@ def generate_code(self, kwargs):
...
-        call_maximinimum = False
-        if len(args) > 1 and isinstance(args[1], (ast.Name, ast.Subscript)):
-            call_maximinimum = True
 
         new_kwargs = self.parse_kwargs(kwargs)
-        if "other" in new_kwargs:
-            call_maximinimum = True
 
-        if call_maximinimum:
+        if len(args) == 2 and len(new_kwargs) == 0:
```
Comment (zhwesky2010) [inline paconvert/api_matcher.py:398]: 这个一刀切不转的话 损失会比较大，我看了下 验证集中，如果第二个参数为变量，就是other用法，如果第二个参数为常量，就是dim用法

所以不用管dim为一个变量的用法，还是要结合实际，可以多结合验证集分析用法

如果第二个参数为变量 或 指定others，则为maximum

其他 全部按 max来做

infoflow 2023-06-15 18-17-27

File: tests/test_max.py [hunk 18]
```diff
@@ -97,21 +82,44 @@ def test_case_6():
...
         import torch
-        dim, keepdim = 1, False
-        result = torch.max(torch.tensor([[1, 2, 3], [3, 4, 6]]), dim, keepdim)
+        result = torch.max(torch.tensor([[1, 2, 3], [3, 4, 6]]), other=torch.tensor([[1, 0, 3], [3, 4, 3]]))
         """
     )
     obj.run(pytorch_code, ["result"])
+
+
+test_case_6()
```
Comment (zhwesky2010) [inline tests/test_max.py:125]: 这个可以删了

File: tests/test_max.py [hunk 19]
```diff
@@ -97,21 +82,44 @@ def test_case_6():
...
+        ["result"],
         unsupport=True,
-        reason="paddle does not return index when dim is specified",
+        reason="Usage cannot be determaxed when there are only two variable parameters, please use key-value param",
     )
 
 
-# paddle does not return index when dim is specified
-def _test_case_7():
+def test_case_9():
```
Comment (zhwesky2010) [inline tests/test_max.py:115]: 加一个没有other关键字的case

File: tests/test_max.py [hunk 20]
```diff
@@ -97,21 +82,44 @@ def test_case_6():
...
     )
+    obj.run(pytorch_code, ["out"])
+
+
+def test_case_7():
+    pytorch_code = textwrap.dedent(
+        """
+        import torch
+        dim, keepdim = 1, False
+        result = torch.max(torch.tensor([[1, 2, 3], [3, 4, 6]]), dim, keepdim)
```
Comment (zhwesky2010) [inline tests/test_max.py:93]: 这种变量为dim的用法看起来没有，可以不用测了。还是要结合实际来做

File: tests/test_max.py [hunk 21]
```diff
@@ -97,21 +82,44 @@ def test_case_6():
...
+        result = torch.max(torch.tensor([[1, 2, 3], [3, 4, 6]]), dim)
+        """
+    )
     obj.run(
         pytorch_code,
-        ["out"],
+        ["result"],
         unsupport=True,
-        reason="paddle does not return index when dim is specified",
+        reason="Usage cannot be determaxed when there are only two variable parameters, please use key-value param",
```
Comment (zhwesky2010) [inline tests/test_max.py:111]: 这个case可以移除

File: tests/test_min.py [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        import torch
+        dim = 1
+        result = torch.min(torch.tensor([[1, 2, 3], [3, 4, 6]]), dim)
+        """
+    )
+    obj.run(
+        pytorch_code,
+        ["result"],
+        unsupport=True,
```
Comment (zhwesky2010) [inline tests/test_min.py:110]: 这个case可以移除

File: tests/test_max.py (hunk 21 above)
Comment (Tomoko-hjf) [reply tests/test_max.py]: done

File: tests/test_max.py (hunk 19 above)
Comment (Tomoko-hjf) [reply tests/test_max.py:109]: done

File: tests/test_min.py (hunk 22 above)
Comment (Tomoko-hjf) [reply tests/test_min.py]: done

**Review by zhwesky2010 — APPROVED** [review]
LGTM
